Source/Rhetos.Utilities/ApplicationConfiguration/ConfigurationSources/RhetosAppEnvironmentSource.cs

```diff
@@ -28,40 +29,30 @@ namespace Rhetos.Utilities.ApplicationConfiguration.ConfigurationSources
     /// </summary>
     public class RhetosAppEnvironmentSource : IConfigurationSource
     {
-        private readonly string _configurationFolder;
-        private readonly JsonFileSource _jsonFileSource;
+        private readonly string _filePath;
 
         public RhetosAppEnvironmentSource(string configurationFolder)
         {
-            _configurationFolder = configurationFolder;
-            _jsonFileSource = new JsonFileSource(Path.Combine(configurationFolder, RhetosAppEnvironment.ConfigurationFileName), optional: true);
+            _filePath = Path.Combine(configurationFolder, RhetosAppEnvironment.ConfigurationFileName);
         }
 
-        public string BaseFolder => _configurationFolder;
-
-        public IDictionary<string, object> Load()
+        public IDictionary<string, IConfigurationValue> Load()
         {
-            var settings = new Dictionary<string, object>(_jsonFileSource.Load());
+            var jsonFileSource = new JsonFileSource(_filePath, optional: true);
+            var settings = new Dictionary<string, IConfigurationValue>(jsonFileSource.Load());
 
             // Settings from configuration generated at build-time:
-
-            string rhetosRuntimePath = settings.GetValueOrDefault(nameof(RhetosAppEnvironment.RhetosRuntimePath)) as string;
+            string rhetosRuntimePath = settings.GetValueOrDefault(nameof(RhetosAppEnvironment.RhetosRuntimePath))?.Value as string;
             if (string.IsNullOrEmpty(rhetosRuntimePath))
-                throw new FrameworkException($"Missing '{nameof(RhetosAppEnvironment.RhetosRuntimePath)}' configuration setting. Please verify that the Rhetos build have passed successfully.");
+                throw new FrameworkException($"Missing '{nameof(RhetosAppEnvironment.RhetosRuntimePath)}' configuration setting. Please verify that Rhetos build has passed successfully.");
 
-            string assetsFolder = settings.GetValueOrDefault(nameof(RhetosAppEnvironment.AssetsFolder)) as string;
+            string assetsFolder = settings.GetValueOrDefault(nameof(RhetosAppEnvironment.AssetsFolder))?.Value as string;
             if (string.IsNullOrEmpty(assetsFolder))
-                throw new FrameworkException($"Missing '{nameof(RhetosAppEnvironment.AssetsFolder)}' configuration setting. Please verify that the Rhetos build have passed successfully.");
-
-            // RhetosAppEnvironment: Converting to full path and adding computed properties.
+                throw new FrameworkException($"Missing '{nameof(RhetosAppEnvironment.AssetsFolder)}' configuration setting. Please verify that Rhetos build has passed successfully.");
 
-            string absolutePath(string relativePath) => relativePath != null ? Path.GetFullPath(Path.Combine(_configurationFolder, relativePath)) : null;
+            settings[nameof(RhetosAppEnvironment.ApplicationRootFolder)] = new FileSourceConfigurationValue("", _filePath);

```

Comment:
Hmm agreed. Entire method was difficult to read for me. If those two are set with verbatim values, then a few more have to be overwritten as well, since we don't want mixed setting members. Right? Not sure about this...

Reply:
Resolved by larger refactor of RhetosAppEnvironment class